src/Structure/Factory/UtilityFactory.php

```diff
@@ -84,7 +85,16 @@ protected function getClass(ReflectionClass $class, Config $config = null, $name
                 $values[] = $config;
             } else if ($config->get($parameter->getName()) !== null) {
                 if ($parameter->getClass()) {
-                    $values[] = $this->getClass($parameter->getClass(), $config->getConfig($parameter->getName()), $parameter->getName());
+                    if ($parameter->getClass()->isInterface()) {
+                        if ($driver = $config->getConfig($parameter->getName())->get('driver')) {
+                            $driverReflection = new ReflectionClass($driver);
+                            $values[] = $this->getClass($driverReflection, $config->getConfig($parameter->getName()), $parameter->getName());
+                        } else {
+                            throw new MissingConfigurationException("Configuration for interface '{$parameter->getName()}' parameter must include a driver");
+                        }
+                    } else {
+                        $values[] = $this->getClass($parameter->getClass(), $config->getConfig($parameter->getName()), $parameter->getName());
```

Comment:
Remove the check for the interface as it may be a class with a subclass as the driver
```php
                        if ($driver = $config->getConfig($parameter->getName())->get('driver')) {
                            $driverReflection = new ReflectionClass($driver);
                            $values[] = $this->getClass($driverReflection, $config->getConfig($parameter->getName()), $parameter->getName());
                        } else {
                            $values[] = $this->getClass($parameter->getClass(), $config->getConfig($parameter->getName()), $parameter->getName());
                    }
```